src/Access/KerberosInit.cpp

```diff
@@ -61,6 +62,11 @@ String KerberosInit::fmtError(krb5_error_code code) const
     return fmt_error;
 }
 
+KerberosInit::KerberosInit()
+{
+    memset(&my_creds, 0, sizeof(my_creds));
```

Comment:
It'd make sense to zero `k5` structure [here](https://github.com/ClickHouse/ClickHouse/pull/56401/files#diff-6afc5f08ac58fa52e17ac2052c0bb16b49a07e1e92f419cbcaf727be9b62e1a6R44) as well

Reply:
I looked into it and `{}` value initialization actually fills the struct with zeroes ([cppref](https://en.cppreference.com/w/cpp/language/value_initialization)), so I did similar thing with `my_creds`